### DISCUSSION TITLE
--kill to forcefully stop running applications

### DISCUSSION THREAD

**Discussion by wsw70** [ScoopInstaller/Scoop#5822] [Ideas]
When running a `scoop update *` I have failures on open/running/locked applications - which is understandable.

It is really painful to scroll the logs to find out which ones failed, to shut them, and restart `scoop update *`.

It would be wonderful to have a `--kill` switch that would forcefully kill running applications (I do no tknow how this works in Windows, in Unix it would be an attempt to "gracefully" stop a process, and then a forced kill afterwards). 

This means that some applications could be killed in a unclean way, while you are using them, ... - but hey, you typed `--kill` knowing you want it. Maybe a confirmation prompt could help here (with the ability to do a `--kill-no-prompt`).

**Comment by rashil2000** [comment]
Force killing of apps has not been implemented yet (for stability reasons), but there's a config value: https://github.com/ScoopInstaller/Scoop/blob/92b71c6057a1a594beeab416cc4d002da17867b1/libexec/scoop-config.ps1#L98-L101 which you can use if you want.

**Reply by wsw70** [reply]
Thank you. I am afraid that without killing the app the issue will still be there because of locked data (I do not know Windows much but this has been my experience).

Maybe in that case a `--dry-run` kind of switch that would show the apps to be updated, with the information about the ones running so that they can be killed ahead of time? In that case I would probably script the kill before updating, which would remove the concern of having scoop do it.

**Reply by rashil2000** [reply]
> Maybe in that case a `--dry-run` kind of switch that would show the apps to be updated, with the information about the ones running so that they can be killed ahead of time? In that case I would probably script the kill before updating, which would remove the concern of having scoop do it.

The default behaviour of Scoop does that already, doesn't it? It lists the running programs and aborts updating them, so that you can kill them yourself and run the update process again.

**Comment by wsw70** [comment]
> The default behaviour of Scoop does that already, doesn't it? It lists the running programs and aborts updating them, so that you can kill them yourself and run the update process again.

Yes, the "problem" (it is just an inconvenience, really) is that I have about 20 or 30 programs installed with scoop and update from time to time. It means that there are screens and screens of output and I am hunting for the red areas to find out which program failed.

My typical process is to update everything, and then update again, wait for the list to appear, abort the update, kill the ones that are listed (these are the ones that could not be updated because they were running), and finally update again.

If there was a way to see it beforehand it would be easier, and if the output is scriptable then all of this could be done in one command.

But again, this is a minor inconvenience, your software is fantastic :)

**Reply by rashil2000** [reply]
I see, I understand your problem now. Yes, the output is cluttered, we have enhancement requests to make it better. Hopefully someone (or me) will get to it someday 😅
